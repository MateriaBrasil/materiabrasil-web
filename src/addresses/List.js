import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id } = props

  return (
    <List
      parentId={id}
      name="addresses"
      path={`/suppliers/${id}/addresses`}
      render={renderList(props)}
    />
  )
}
