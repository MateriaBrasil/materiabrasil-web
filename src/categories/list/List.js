import React, { Component, Fragment } from 'react'
import Typography from '@material-ui/core/Typography'
import isEmpty from 'lodash/isEmpty'
import Button from '@material-ui/core/Button'

import fullName from 'categories/fullName'
import renderCategory from 'categories/renderCategory'
import handleChangeExpanded from 'categories/handleChangeExpanded'

export default class extends Component {
  constructor(props) {
    super(props)
    this.state = { expandedCategories: [] }
    this.onChangeExpanded = handleChangeExpanded.bind(this)
  }

  render() {
    const { list, filters } = this.props
    const { selectedCategories } = filters

    const props = {
      ...this.props,
      ...this.state,
      onChangeExpanded: this.onChangeExpanded,
    }
    console.log(selectedCategories, this.props)

    return (
      <Fragment>
        {!isEmpty(selectedCategories) && (
          <div style={{ marginBottom: 30 }}>
            <Typography variant="subheading" style={{ marginBottom: 20 }}>
              Filtros selecionados
            </Typography>
            {selectedCategories.map(category => {
              console.log('category:',category)
              return (
                <Typography key={category.id} variant="body1">
                  {fullName(list, category, category.name)}
                </Typography>
              )
            })}
            <Button
              style={{ marginTop: 20 }}
              variant="outlined"
              onClick={filters.actions.reset}
            >
              Limpar filtros
            </Button>
          </div>
        )}
        {list.map(renderCategory(props))}
      </Fragment>
    )
  }
}
