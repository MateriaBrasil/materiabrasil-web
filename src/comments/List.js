import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id, type } = props

  return (
    <List
      parentId={id}
      name="comments"
      path={`/${type}/${id}/comments`}
      render={renderList(props)}
    />
  )
}
