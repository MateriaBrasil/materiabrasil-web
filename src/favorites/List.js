import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id, parentId } = props

  return (
    <List
      parentId={parentId}
      name="favorites"
      path={`/albums/${id}/favorites`}
      render={renderList(props)}
    />
  )
}
