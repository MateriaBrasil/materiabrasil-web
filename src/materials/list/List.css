.materials-list-grid {
  margin-top: 10px !important;
}

@media (min-width: 960px)
{
  .materials-list-grid {
    margin-top: 20px !important;
  }
}

@media (min-width: 1280px)
{
  .materials-list-grid {
    margin-top: -40px !important;
  }
}
