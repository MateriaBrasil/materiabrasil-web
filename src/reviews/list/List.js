import React, { Component } from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import withMobileDialog from '@material-ui/core/withMobileDialog'
import Link from 'react-router-dom/Link'
import Button from '@material-ui/core/Button'

import Reviews from './Reviews'

class List extends Component {
  render() {
    const { fullScreen, id, list } = this.props

    return (
      <Dialog
        fullScreen={fullScreen}
        open
        aria-labelledby="responsive-dialog-title"
        maxWidth="md"
        fullWidth
      >
        <DialogTitle id="responsive-dialog-title">Avaliações</DialogTitle>
        <DialogContent>
          <Reviews list={list} />
        </DialogContent>
        <DialogActions>
          <Link to={`/${id}`} style={{ textDecoration: 'none' }}>
            <Button color="primary">Fechar</Button>
          </Link>
        </DialogActions>
      </Dialog>
    )
  }
}

export default withMobileDialog()(List)
