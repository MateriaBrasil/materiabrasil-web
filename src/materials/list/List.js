import React, { Component } from 'react'
import isEmpty from 'lodash/isEmpty'
import Grid from '@material-ui/core/Grid'

import Error from 'Error'
import renderIcons from './renderIcons'
import Card from './Card'
import './List.css'

export default class extends Component {
  componentDidUpdate(prevProps) {
    const { location, listProps } = this.props
    const { location: prevLocation } = prevProps

    if (listProps && location.search !== prevLocation.search) {
      const { actions } = listProps
      actions.fetchList()
    }
  }

  render() {
    const { list, currentUser } = this.props

    if (isEmpty(list)) {
      return <Error color="default">Nenhum material encontrado.</Error>
    }

    return (
      <Grid item xs={12} md={9} className="materials-list-grid">
        <Grid container spacing={8}>
          {list.map(material => {
            const { id } = material
            return (
              <Card
                key={id}
                material={material}
                renderIcons={renderIcons({ currentUser, id, material })}
                {...this.props}
              />
            )
          })}
        </Grid>
      </Grid>
    )
  }
}
