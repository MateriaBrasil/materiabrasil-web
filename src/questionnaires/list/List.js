import React from 'react'
import map from 'lodash/map'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import FormGroup from '@material-ui/core/FormGroup'
import FormControl from '@material-ui/core/FormControl'
import RadioGroup from '@material-ui/core/RadioGroup'
import Radio from '@material-ui/core/Radio'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Typography from '@material-ui/core/Typography'

export default ({ list }) => (
  <List>
    {map(list, questionnaire => {
      const { name, questions } = questionnaire

      return (
        <ListItem key={name}>
          <Typography variant="headline">{name}</Typography>
          {map(questions, question => {
            const { options, description } = question

            return (
              <FormGroup key={description}>
                <Typography variant="subheading">{description}</Typography>
                <FormControl component="fieldset">
                  <RadioGroup
                    key={description}
                    aria-label="Questionnarie Options"
                    name="questionnaire-options"
                  >
                    {map(options, option => (
                      <FormControlLabel
                        value={toString(option.id)}
                        key={option.id}
                        disabled
                        control={<Radio />}
                        label={option.description}
                      />
                    ))}
                  </RadioGroup>
                </FormControl>
              </FormGroup>
            )
          })}
        </ListItem>
      )
    })}
  </List>
)
