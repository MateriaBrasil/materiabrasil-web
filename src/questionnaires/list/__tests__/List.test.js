import React from 'react'
import renderer from 'react-test-renderer'
import List from '../List'

const list = [
  {
    questions: [
      {
        descripton: 'question1',
        options: [
          {
            id: 1,
            description: 'option10',
          },
          {
            id: 2,
            description: 'option11',
          },
        ],
      },
      {
        descripton: 'question2',
        options: [
          {
            id: 1,
            description: 'option20',
          },
          {
            id: 2,
            description: 'option21',
          },
        ],
      },
    ],
    name: 'Foo text',
  },
]

it('renders correctly', () => {
  const tree = renderer.create(<List list={list} />).toJSON()
  expect(tree).toMatchSnapshot()
})
