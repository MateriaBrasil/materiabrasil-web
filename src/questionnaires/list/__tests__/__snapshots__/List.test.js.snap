// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-labelledby="responsive-dialog-title"
  fullScreen={false}
  fullWidth={true}
  onBackdropClick={[Function]}
  onEscapeKeyDown={[Function]}
  open={true}
>
  Dialog - 
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogTitle-root-1"
      id="responsive-dialog-title"
    >
      <h2
        className="MuiTypography-root-2 MuiTypography-title-8"
      >
        Responder question√°rios
      </h2>
    </div>
    <div
      className="MuiDialogContent-root-37"
    >
      <ul
        className="MuiList-root-38 MuiList-padding-39"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <li
          className="MuiListItem-root-42 MuiListItem-default-45 MuiListItem-gutters-49"
          disabled={false}
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <h5
            className="MuiTypography-root-2 MuiTypography-h5-18"
          >
            Foo text
          </h5>
          <div
            className="MuiFormGroup-root-53"
          >
            <h6
              className="MuiTypography-root-2 MuiTypography-subtitle1-20"
            >
              question1
            </h6>
            <fieldset
              className="MuiFormControl-root-55"
            >
              <div
                aria-label="Questionnarie Options"
                className="MuiFormGroup-root-53"
                role="radiogroup"
              >
                <label
                  className="MuiFormControlLabel-root-59 MuiFormControlLabel-disabled-61"
                >
                  <span
                    className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiIconButton-root-72 MuiIconButton-disabled-76 MuiSwitchBase-root-68 MuiRadio-root-63 MuiRadio-colorSecondary-67 MuiSwitchBase-disabled-70 MuiRadio-disabled-65"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex="-1"
                  >
                    <span
                      className="MuiIconButton-label-77"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-81"
                        color={undefined}
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <input
                        autoFocus={undefined}
                        checked={false}
                        className="MuiSwitchBase-input-71"
                        disabled={true}
                        id={undefined}
                        name="questionnaire-options"
                        onChange={[Function]}
                        readOnly={undefined}
                        required={undefined}
                        tabIndex={undefined}
                        type="radio"
                        value="[object Window]"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root-2 MuiTypography-body1-11 MuiFormControlLabel-label-62 MuiFormControlLabel-disabled-61"
                  >
                    option10
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root-59 MuiFormControlLabel-disabled-61"
                >
                  <span
                    className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiIconButton-root-72 MuiIconButton-disabled-76 MuiSwitchBase-root-68 MuiRadio-root-63 MuiRadio-colorSecondary-67 MuiSwitchBase-disabled-70 MuiRadio-disabled-65"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex="-1"
                  >
                    <span
                      className="MuiIconButton-label-77"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-81"
                        color={undefined}
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <input
                        autoFocus={undefined}
                        checked={false}
                        className="MuiSwitchBase-input-71"
                        disabled={true}
                        id={undefined}
                        name="questionnaire-options"
                        onChange={[Function]}
                        readOnly={undefined}
                        required={undefined}
                        tabIndex={undefined}
                        type="radio"
                        value="[object Window]"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root-2 MuiTypography-body1-11 MuiFormControlLabel-label-62 MuiFormControlLabel-disabled-61"
                  >
                    option11
                  </span>
                </label>
              </div>
            </fieldset>
          </div>
          <div
            className="MuiFormGroup-root-53"
          >
            <h6
              className="MuiTypography-root-2 MuiTypography-subtitle1-20"
            >
              question2
            </h6>
            <fieldset
              className="MuiFormControl-root-55"
            >
              <div
                aria-label="Questionnarie Options"
                className="MuiFormGroup-root-53"
                role="radiogroup"
              >
                <label
                  className="MuiFormControlLabel-root-59 MuiFormControlLabel-disabled-61"
                >
                  <span
                    className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiIconButton-root-72 MuiIconButton-disabled-76 MuiSwitchBase-root-68 MuiRadio-root-63 MuiRadio-colorSecondary-67 MuiSwitchBase-disabled-70 MuiRadio-disabled-65"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex="-1"
                  >
                    <span
                      className="MuiIconButton-label-77"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-81"
                        color={undefined}
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <input
                        autoFocus={undefined}
                        checked={false}
                        className="MuiSwitchBase-input-71"
                        disabled={true}
                        id={undefined}
                        name="questionnaire-options"
                        onChange={[Function]}
                        readOnly={undefined}
                        required={undefined}
                        tabIndex={undefined}
                        type="radio"
                        value="[object Window]"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root-2 MuiTypography-body1-11 MuiFormControlLabel-label-62 MuiFormControlLabel-disabled-61"
                  >
                    option20
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root-59 MuiFormControlLabel-disabled-61"
                >
                  <span
                    className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiIconButton-root-72 MuiIconButton-disabled-76 MuiSwitchBase-root-68 MuiRadio-root-63 MuiRadio-colorSecondary-67 MuiSwitchBase-disabled-70 MuiRadio-disabled-65"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex="-1"
                  >
                    <span
                      className="MuiIconButton-label-77"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-81"
                        color={undefined}
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <input
                        autoFocus={undefined}
                        checked={false}
                        className="MuiSwitchBase-input-71"
                        disabled={true}
                        id={undefined}
                        name="questionnaire-options"
                        onChange={[Function]}
                        readOnly={undefined}
                        required={undefined}
                        tabIndex={undefined}
                        type="radio"
                        value="[object Window]"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root-2 MuiTypography-body1-11 MuiFormControlLabel-label-62 MuiFormControlLabel-disabled-61"
                  >
                    option21
                  </span>
                </label>
              </div>
            </fieldset>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="MuiDialogActions-root-90"
    >
      <button
        className="MuiButtonBase-root-78 MuiButton-root-92 MuiButton-text-94 MuiButton-textPrimary-95 MuiButton-flat-97 MuiButton-flatPrimary-98 MuiDialogActions-action-91"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-93"
        >
          Fechar
        </span>
        <span
          className="MuiTouchRipple-root-118"
        />
      </button>
      <button
        className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiButton-root-92 MuiButton-text-94 MuiButton-textPrimary-95 MuiButton-flat-97 MuiButton-flatPrimary-98 MuiButton-disabled-112 MuiDialogActions-action-91"
        disabled={true}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="submit"
      >
        <span
          className="MuiButton-label-93"
        >
          Salvar
        </span>
      </button>
    </div>
  </form>
</div>
`;
